metaflow/plugins/argo/argo_client.py

```diff
@@ -203,3 +203,146 @@ def schedule_workflow_template(self, name, schedule=None):
             raise ArgoClientException(
                 json.loads(e.body)["message"] if e.body is not None else e.reason
             )
+
+    def register_trigger_on_template(self, name, sensor_template):
+        try:
+            sensor_template["metadata"][
+                "resourceVersion"
+            ] = self._client.CustomObjectsApi().get_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="sensors",
+                name=name,
+            )[
+                "metadata"
+            ][
+                "resourceVersion"
+            ]
+        except self._client.rest.ApiException as e:
+            if e.status == 404:
+                try:
+                    return (
+                        self._client.CustomObjectsApi().create_namespaced_custom_object(
+                            group=self._group,
+                            version=self._version,
+                            namespace=self._namespace,
+                            plural="sensors",
+                            body=sensor_template,
+                        )
+                    )
+                except self._client.rest.ApiException as e:
+                    raise ArgoClientException(
+                        json.loads(e.body)["message"]
+                        if e.body is not None
+                        else e.reason
+                    )
+            else:
+                raise ArgoClientException(
+                    json.loads(e.body)["message"] if e.body is not None else e.reason
+                )
+        try:
+            return self._client.CustomObjectsApi().replace_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="sensors",
+                name=name,
+                body=sensor_template,
+            )
+        except self._client.rest.ApiException as e:
+            raise ArgoClientException(
+                json.loads(e.body)["message"] if e.body is not None else e.reason
+            )
+
+    def remove_existing_trigger_on_template(self, name, dry_run=False):
```

Comment:
If we drop `explicit deletes` for now, we should be able to drop these methods as well